dbt/contracts/connection.py

```diff
@@ -24,6 +23,18 @@
     Required('schema'): basestring,
 })
 
+redshift_auth_methods = ['database', 'iam']
+redshift_credentials_contract = Schema({
+    Required('method'): Any(*redshift_auth_methods),
```

Comment:
I think this should be optional, and the default should be `database`. It would be good to preserve backwards compatibility here

Reply:
Roger that. It would be great to require a password if the `method` keyword is missing or the `database` method is selected. Similarly, for `cluster_id` if the `iam` method is used. How do I go about doing so with voluptuous?

Reply:
We're in the process of ripping out voluptuous and replacing it with json schemas as a part of building out [dbt's API](https://github.com/fishtown-analytics/dbt/compare/dev/betsy-ross#diff-b7e9e1173454a0ea41bb686f9dc1bb2bR4). I think you can just leave the voluptuous code here and we'll convert it to the new style contract along with everything else